internal/consensus/state.go

```diff
@@ -1310,8 +1311,21 @@ func (cs *State) createProposalBlock(ctx context.Context) (*types.Block, error)
 		lastExtCommit = &types.ExtendedCommit{}
 
 	case cs.LastCommit.HasTwoThirdsMajority():
-		// Make the commit from LastCommit
-		lastExtCommit = cs.LastCommit.MakeExtendedCommit(cs.state.ConsensusParams.Feature)
+		// Make the commit from LastCommit.
+		//
+		// Note we can't aggregate a commit when vote extensions are enabled
+		// because votes are different.
```

Comment:
Hmmm, I might be wrong, but I don't agree with this.
When vote extensions are enabled, there is one signature for the vote and another signature for the vote extension.
I agree the extension signatures cannot be aggregated, as the extensions can be different.
But the vote themselves _can_ still be aggregated (assuming you remove the timestamp field, or set it always to 0-time).

Reply:
Good point. But then again, who is going to verify vote extensions? There's no point in aggregating votes if you need to verify vote extensions using individual verification (signature.Verify).

Reply:
On the 2nd thought, we don't verify vote extensions in `VerifyCommit`, so I think using them while aggregating vote signatures is okay. [`6b0b06e` (#11)](https://github.com/berachain/cometbft/pull/11/commits/6b0b06e5e707d829a88871fbcf65ce7142b664ac)